packages/components/src/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { ReactElement, useState } from "react";
+import classnames from "classnames";
+import ReactDatePicker from "react-datepicker";
+/**
+ * Disabling no-internal-modules here because we need
+ * to reach into the package to get the css file.
+ */
+// eslint-disable-next-line import/no-internal-modules
+import "react-datepicker/dist/react-datepicker.css";
+import { XOR } from "ts-xor";
+import styles from "./DatePicker.css";
+import { DatePickerCustomHeader } from "./DatePickerCustomHeader";
+import { strFormatDate } from "../FormatDate";
+import { Button } from "../Button";
+
+export interface DatePickerReturnedDates {
+  readonly formatted: string;
+  readonly raw: Date;
+}
+
+interface BaseDatePickerProps {
+  /**
+   * Change handler that will return the date selected.
+   */
+  onChange(val: DatePickerReturnedDates): void;
+}
+
+interface DatePickerModalProps extends BaseDatePickerProps {
+  /**
+   * Use a custom activator to trigger the DatePicker
+   */
+  readonly activator?: ReactElement;
+}
+
+interface DatePickerInlineProps extends BaseDatePickerProps {
+  readonly inline?: boolean;
+}
+
+type DatePickerProps = XOR<DatePickerModalProps, DatePickerInlineProps>;
+
+export function DatePicker({ onChange, activator, inline }: DatePickerProps) {
+  const [startDate, setStartDate] = useState(new Date());
+
+  const datePickerClassNames = classnames(styles.datePicker, {
+    [styles.inline]: inline,
+  });
+
+  return (
+    <div>
```

Comment:
Might be best if this is inline-block

Reply:
also maybe a `<Fragment />` if `inline` is true.

```
import React, { Fragment, ReactElement, useState } from "react";

const Tag = inline ? "Fragment" : "div";

return <Tag>{...}</Tag>
```

Or look at how the element is being laid out in the browser and make em clearer and _intentional_, it seems inconsistent on what it would do. Right now it's
- Taking full width when there's an activator (even tho the button won't take the full width, the parent does)
- Shrink to it's smallest size when it's inline (same deal as above ^ but since we wrapped it in the div, the parent is taking full width)